modules/ROOT/examples/live-demos/custom-toolbar-menu-button-search/index.js

```diff
@@ -0,0 +1,53 @@
+tinymce.init({
+  selector: 'textarea#custom-toolbar-menu-button-search',
+  height: 500,
+  toolbar: 'mybutton',
+
+  setup: (editor) => {
+    /* Menu items are recreated when the menu is closed and opened, so we need
+       a variable to store the toggle menu item state. */
+    let toggleState = false;
+
+    /* example, adding a toolbar menu button */
+    editor.ui.registry.addMenuButton('mybutton', {
+      text: 'My searchable button',
+      search: {
+        placeholder: 'What do you want to search for?'
+      },
+      fetch: (callback, fetchContext) => {
+        if (fetchContext.pattern.length > 0) {
+          callback([
+            {
+              type: 'menuitem',
+              text: `You searched for: "${fetchContext.pattern}"`,
+              onAction: () => editor.insertContent(`<strong>Inserted selected search result</strong>`)
+            }
+          ])
+        } else {
+          const items = [
```

Comment:
It might be confusing that I removed the nestedmenu items here, and it may cause readers to think that they aren't supported. I mainly simplified the example to focus on what was different about it. I'm not sure if we need to explicitly mention that nestedmenus are supported, or whether it is sufficient to just not mention that they aren't.

Reply:
Might be confusing to include them in the same button, I think mentioning that they are supported should suffice. I feel that this fetch callback should include a few example search items though, rather than just a single action for anything searched.

Reply:
Yeah, it's the difference between showing a typical use case versus showing how the button works. At the moment, it's just showing how the button works. Searching is the typical use case, but it is possible to use it for other things.

Reply:
See: https://github.com/tinymce/tinymce-docs/pull/2520#discussion_r976460331